src/causalprog/causal_problem.py

```diff
@@ -0,0 +1,182 @@
+"""Extension of the Graph class providing features for solving causal problems."""
+
+from collections.abc import Callable
+from inspect import signature
+from typing import TypeAlias
+
+import jax
+import jax.numpy as jnp
+
+from causalprog._abc.labelled import Labelled
+from causalprog.graph import Graph, Node
+
+CausalEstimand: TypeAlias = Callable[..., float]
+Constraints: TypeAlias = Callable[..., float]
+
+
+def raises(exception: Exception) -> Callable[[], float]:
+    """Create a callable that raises ``exception`` when called."""
+
+    def _inner() -> float:
+        raise exception
+
+    return _inner
+
+
+class CausalProblem(Labelled):
+    """f."""
```

Comment:
This looks like a placeholder docstring

Reply:
You know when you really want ruff to stop telling you about something but forget to go back to it? :sweat_smile:

Have filled this in now. Pre-emptively with info about the constraints function too, since I'll be adding that next.